crates/next-core/src/next_font/google/options.rs

```diff
@@ -170,8 +170,17 @@ pub(super) fn options_from_request(
     }
 
     if let Some(axes) = argument.axes.as_ref() {
-        if !axes.is_empty() && !matches!(weights, FontWeights::Variable) {
-            return Err(anyhow!("Axes can only be defined for variable fonts"));
+        if !axes.is_empty() {
+            if !matches!(weights, FontWeights::Variable) {
+                return Err(anyhow!("Axes can only be defined for variable fonts."));
+            }
+
+            if weights.get(0) != Some(&FontWeights::Variable) {
+                return Err(anyhow!(
+                    "Axes can only be defined for variable fonts when the weight property is \
+                     nonexistent or set to `variable`."
+                ));
```

Comment:
The anyhow crate has a shorthand `bail!` macro: https://docs.rs/anyhow/latest/anyhow/macro.bail.html

```suggestion
                anyhow::bail!(
                    "Axes can only be defined for variable fonts when the weight property is \
                     nonexistent or set to `variable`."
                );
```

Reply:
I see this is just matching the pattern of the rest of the function. I'll see if I can follow up with a low-effort astgrep codemod after this.